droid-api/src/test/java/uk/gov/nationalarchives/droid/internal/api/DroidAPITest.java

```diff
@@ -109,23 +113,38 @@ public void testFileExtensions() throws IOException, SignatureParseException {
     }
 
     @Test
-    public void testContainerVersions() throws SignatureParseException {
-        DroidAPI api = DroidAPITestUtils.createApi();
+    public void should_report_all_puids_when_there_are_more_than_one_identification_hits() throws IOException {
+        List<ApiResult> results = api.submit(Paths.get("src/test/resources/double-identification.jpg"));
+        assertThat(results.size(), is(2));
+        assertThat(results.stream().map(ApiResult::getPuid).collect(Collectors.toList()),
+                containsInAnyOrder("fmt/96", "fmt/41"));
+        assertThat(results.stream().map(ApiResult::getName).collect(Collectors.toList()),
+                containsInAnyOrder("Raw JPEG Stream", "Hypertext Markup Language"));
+    }
+
+    @Test
+    public void should_report_when_there_is_an_extension_mismatch() throws IOException {
+        List<ApiResult> results = api.submit(Paths.get("src/test/resources/docx-file-as-xls.xlsx"));
+        assertThat(results.size(), is(1));
+        assertThat(results.get(0).getPuid(), is("fmt/412"));
+        assertThat(results.get(0).isFileExtensionMismatch(), is(true));
+    }
+
+    @Test
+    public void should_report_correct_version_for_the_binary_and_container_signature() {
         assertThat(api.getContainerSignatureVersion(), is("20221102"));
         assertThat(api.getDroidVersion(), is(ResourceBundle.getBundle("options").getString("version_no")));
         assertThat(api.getBinarySignatureVersion(), is("109"));
     }
 
     @Test
-    public void testEmptyFileWithNoExtension() throws IOException, SignatureParseException {
-        DroidAPI api = DroidAPITestUtils.createApi();
+    public void should_report_correct_size_of_zero_for_an_empty_file() throws IOException {
```

Comment:
This is just a suggestion but is it worth changing this test name slightly. `size_of_zero` sounds like you're referring to the file size. Maybe something like `should_produce_zero_results_for_an_empty_file` ? Then it matches the test name below as well.

Reply:
Done